File: CivitasJava/civitas/Casilla_Sorpresa.java
package civitas;

import java.util.ArrayList;

public class Casilla_Sorpresa extends Casilla
{
    private MazoSorpresas mazo;
    private Sorpresa sorpresa;
    
    Casilla_Sorpresa(String nombre,MazoSorpresas mazo) { 

        super(nombre);
        this.mazo = mazo;
        this.sorpresa=null;
    }
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos) {

        if (super.jugadorCorrecto(iactual, todos)) 
        {
            this.sorpresa = this.mazo.siguiente();
            this.sorpresa.aplicarAJugador(iactual, todos);
        }
    }
    @Override 
    public String toString()
    {
        String ret=super.getNombre();
        ret += " de tipo sorpresa : \n";
        if(this.sorpresa!=null){
            ret += this.sorpresa.toString();
        }
        return ret;
    }
}

File: CivitasRuby/Casilla_Sorpresa.rb
module Civitas
    class Casilla_Sorpresa < Casilla
        def initialize(nombre,mazo)
            super(nombre)
            @mazo=mazo
        end

        def recibeJugador(actual,todos)

            if jugadorCorrecto(actual, todos)
                @sorpresa=@mazo.siguiente
                informe(actual, todos)
                @sorpresa.aplicarAJugador(actual, todos)
            end
        end

        def toString
            ret=@nombre + ' de tipo sorpresa: '
            ret+=@sorpresa.toString
            return ret
        end

    end
end